include/tvm/driver/driver_api.h

```diff
@@ -78,20 +78,22 @@ TVM_DLL transform::Sequential HostModulePassManager(IRModule mixed_mod, Target t
  * \brief Lower an IRModule (optimize with it with the pass list defined in CreatePassList)
  * \param mod The IRmodule to lower
  * \param simple_mode Disables the loop partition pass. Defaults to false.
+ * \param target The device target to provide additional characteristics. Defaults to not defined.
  * \return The result module.
  */
-TVM_DLL IRModule LowerModule(IRModule mod, bool simple_mode = false);
+TVM_DLL IRModule LowerModule(IRModule mod, bool simple_mode = false, Target target = {});
```

Comment:
I have some doubts about introducing the new parameter with target to many lowering functions. As far as I understood, the only one thing why we need to have this `target` parameter is for passing value of `vtcm_capacity` to the lowering functions.
From my point of view, we shouldn't modify this common API. As for me, the better way is to pass this option through `PassContext` (e.g. in the same way it works for `relay.ToMixedPrecision.keep_orig_output_dtype`).

@tqchen, @csullivan, @junrushao, @tkonolige, In file [test_vtcm.py](https://github.com/apache/tvm/pull/13349/files#diff-9380a067fbc7e640cb0257906fadaf5c400f07f46500d530af7c601b29855eb6R83-R95) @Icemist has presented possible ways for passing value of `vtcm_capacity`:
1. By passing target as a parameter to the `lower` function: 
    ```python
    _raises_exception(lambda: tvm.lower(sch.mod["main"], target=target)) == limited
    ``` 
2. By using `with` statement:
    ```python
    with target:
        assert (
            _raises_exception(lambda: tvm.lower(sch.mod["main"])) == limited
        ), "VTCM memory allocation limiter does not work correctly "
    ```
3. By passing value of `vtcm_capacity` to `PassContext`:
    ```python
    with tvm.transform.PassContext(config={"tir.vtcm_capacity": vtcm_capacity}):
        assert (
            _raises_exception(lambda: tvm.lower(sch.mod["main"])) == limited
        ), "VTCM memory allocation limiter does not work correctly "
    ```
    
Personally, I prefer the third option, because in this case, I believe, it is not necessary to modify the API of lowering functions. What do you think about that?

Reply:
In the case of a refusal to change the api and using only tir.vtcm_capacity as a transmitter - 

In tvm.build we can have this logic:
```python
vtcm_capacity = target.vtcm_capacity if target.kind.name == "hexagon" else 0
with tvm.transform.PassContext(config={"tir.vtcm_capacity": vtcm_capacity}):
	# code uses lowering
	
```
to leave the API as pleasant as possible:
```python
target = tvm.target.hexagon("v68", vtcm_capacity=128)
target = tvm.target.Target(target, host=target)
func = tvm.build(sch, [], target, name="func")
```

But also anticipate that it will be necessary to find and test/modify additional functions including C++ and python parts that use the lowering for the hexagon target.

Reply:
For now we should avoid making an API change to the compiler that would introduce `target` to lowering without some community discussion on the topic.

That said I also prefer deriving this target specific information from the target. In this case if we make the following change to move LowerVTCMAlloc into the MixedModulePassManager, we can add `tir::transform::VerifyVTCMLimit` just prior to it in the `mixed_pass_list`. As these passes are run at build time (after lowering), the target is available for use. 

```diff
diff --git a/src/driver/driver_api.cc b/src/driver/driver_api.cc
index e5e3998b1..babac2bf6 100644
--- a/src/driver/driver_api.cc
+++ b/src/driver/driver_api.cc
@@ -225,10 +225,8 @@ Array<tvm::transform::Pass> CreatePassList(bool disable_loop_partition) {
   if (!disable_storage_rewrite) {
     pass_list.push_back(tir::transform::StorageRewrite());
   }
-  // LowerVtcmAlloc must occur after any transformations that modify memory allocation locations
-  pass_list.push_back(tir::transform::LowerVtcmAlloc());
-  bool use_async_copy = pass_ctx->GetConfig<Bool>("tir.use_async_copy", Bool(false)).value();

+  bool use_async_copy = pass_ctx->GetConfig<Bool>("tir.use_async_copy", Bool(false)).value();
   if (use_async_copy) {
     pass_list.push_back(tir::transform::LowerAsyncDMA());
   }
@@ -539,6 +537,9 @@ transform::Sequential MixedModulePassManager(IRModule mixed_mod, Target target)

   Array<Pass> mixed_pass_list;
+  mixed_pass_list.push_back(tir::transform::VerifyVTCMLimit(GetVTCMCapacity(target, pass_ctx)));
+  // LowerVtcmAlloc must occur after any transformations that modify memory allocation locations
+  mixed_pass_list.push_back(tir::transform::LowerVtcmAlloc());
+
   mixed_pass_list.push_back(tir::transform::BindTarget(target));

   mixed_pass_list.push_back(tir::transform::VerifyMemory());
```